File: tcs_manager_system/src/main/java/com/tcsManager/tcsmanager/service/TesterService.java
package com.tcsManager.tcsmanager.service;

import java.util.List;
import java.util.Map;

import com.tcsManager.tcsmanager.entity.Tester;
import com.tcsManager.tcsmanager.exception.TestCaseNotFoundException;
import com.tcsManager.tcsmanager.exception.TesterNotFoundException;

import org.springframework.http.ResponseEntity;

/**
 * TesterService
 */
public interface TesterService {

    ResponseEntity<Tester> getTester(long testerId) throws TesterNotFoundException;

    ResponseEntity<Tester> createTester(Tester tester, long testCaseId) throws TestCaseNotFoundException;

    ResponseEntity<List<Tester>> getAllTesters();

    ResponseEntity<Tester> updateTester(long testerId, Tester testerUpdated) throws TesterNotFoundException;

    ResponseEntity<Tester> partialUpdateTester(long testerId, Map<Object, Object> fields)
            throws TesterNotFoundException;

    ResponseEntity<?> deleteTester(long testerId, long testCaseId)
            throws TesterNotFoundException, TestCaseNotFoundException;

}

File: tcs_manager_frontend/src/service/TesterService.js
import axios from "axios";

const TESTER_BASE_API_URL = "http://localhost:8080/api/v1/testers/";

class TesterService {

    getAllTesters() {
        return axios.get(TESTER_BASE_API_URL)
            .then(response => response.data)
    }

    getTester(testerId) {
        return axios.get(TESTER_BASE_API_URL + `${testerId}`)
            .then(response => response.data)
    }

    createTester(testCaseId, tester) {
        return axios.post(TESTER_BASE_API_URL + `tc/${testCaseId}`, tester)
            .then(response => response.data)
    }

    updateTester(testerId) {
        return axios.put(TESTER_BASE_API_URL + `${testerId}`)
            .then(response => response.data)
    }

    partialUpdateTester(testerId, tester) {
        return axios.patch(TESTER_BASE_API_URL + `${testerId}`, tester)
            .then(response => response.data)
    }

    deleteTester(testerId, testCaseId) {
        return axios.delete(TESTER_BASE_API_URL + `${testerId}/tc/${testCaseId}`)
            .then(response => response.data)
    }
}

export default new TesterService();
